### PR TITLE
CRD changes for Syncthing data mover

### PR DESCRIPTION
**Pull request by RobotSail**
Signed-off-by: Oleg <97077423+RobotSail@users.noreply.github.com>

**Describe what this PR does**

**Is there anything that requires special attention?**

**Related issues:**

#19

### CODE REVIEW

File: api/v1alpha1/replicationdestination_types.go [hunk 1]
```diff
@@ -200,6 +209,12 @@ type ReplicationDestinationResticSpec struct {
 	RestoreAsOf *string `json:"restoreAsOf,omitempty"`
 }
 
+type ReplicationDestinationSyncthingStatus struct {
+	// NodeList is a list of the Syncthing nodes we are currently connected to.
+	NodeList []*SyncthingNode `json:"nodeList,omitempty"`
+	DeviceID string           `json:"deviceID,omitempty"`
+}
```
Comment (JohnStrunk) [inline api/v1alpha1/replicationdestination_types.go:216]:
```suggestion
type ReplicationDestinationSyncthingStatus struct {
	Address string `json:"address,omitempty"`
	DeviceID string           `json:"deviceID,omitempty"`
	// NodeList is a list of the Syncthing nodes we are currently connected to.
	connectedPeers []SyncthingPeer `json:"connectedPeers,omitempty"`
}
```

**Comment by JohnStrunk** [comment]
/retitle CRD changes for Syncthing data mover

File: api/v1alpha1/replicationsource_types.go [hunk 2]
```diff
@@ -223,6 +234,13 @@ type ReplicationSourceRsyncStatus struct {
 	Port *int32 `json:"port,omitempty"`
 }
 
+type ReplicationSourceSyncthingStatus struct {
+	// NodeList is a list of the Syncthing nodes we are currently connected to.
+	NodeList []SyncthingPeer `json:"nodeList,omitempty"`
+	// Device ID of the current syncthing device
+	DeviceID string `json:"deviceID,omitempty"`
+}
```
[the remark this answers is not in the document]
Comment (RobotSail) [reply api/v1alpha1/replicationsource_types.go:249]: Added

File: api/v1alpha1/replicationsource_types.go [hunk 3]
```diff
@@ -180,6 +180,14 @@ type ReplicationSourceResticStatus struct {
 	LastPruned *metav1.Time `json:"lastPruned,omitempty"`
 }
 
+// define the Syncthing field
+type ReplicationSourceSyncthingSpec struct {
```
[the remark this answers is not in the document]
Comment (RobotSail) [reply api/v1alpha1/replicationsource_types.go:184]: `Introducer` is a peer setting, so if `A` is configuring `B` as a connection, `A` would need to set `B` as an introducer to automatically add `B`'s other connections.

[the remark this answers is not in the document]
Comment (JohnStrunk) [reply api/v1alpha1/replicationsource_types.go:184]: Good catch. I guess it belongs in the SyncthingPeer structure.

File: api/v1alpha1/replicationsource_types.go [hunk 4]
```diff
@@ -223,6 +234,15 @@ type ReplicationSourceRsyncStatus struct {
 	Port *int32 `json:"port,omitempty"`
 }
 
+type ReplicationSourceSyncthingStatus struct {
+	// NodeList is a list of the Syncthing nodes we are currently connected to.
+	PeerList []SyncthingPeerStatus `json:"peerList,omitempty"`
```
Comment (JohnStrunk) [inline api/v1alpha1/replicationsource_types.go:239]: Peers, like above?

**Review by JohnStrunk — COMMENTED** [review]
Looking good... just "Peers" vs "PeerList"

**Comment by JohnStrunk** [comment]
/approve

@tesshuflower, I'm ready to merge this. Do you want to take a look?

**Comment by tesshuflower** [comment]
> /approve
> 
> @tesshuflower, I'm ready to merge this. Do you want to take a look?

I think it looks good to me - Just one thing thing (Mostly for my understanding) @RobotSail  - in the SyncthingStatus there is a list of peer status that has things like `connected` - will the controller be able to determine that in order to fill out in the status? or is that something only known by the syncthing mover job itself.

**Comment by RobotSail** [comment]
@tesshuflower The Syncthing mover job is what actually stores & updates that information, the controller simply obtains it by querying the mover's API.

**Review by tesshuflower — COMMENTED** [review]
/lgtm

**Pull request merged**